src/lib/loaders/loaders_without_authentication/gravity.ts

```diff
@@ -27,6 +28,8 @@ export default opts => {
   }
 
   return {
+    featureLoader: (id: string) =>
+      gravityLoader(id => `feature/${id}`)(id).then(Gravity.Feature.check),
```

Comment:
✔️

Reply:
😮 !

Reply:
@dzucconi - mind briefly explaining this form / what it does?

Reply:
@damassi sure — basically just compares the response to the type we're expecting at runtime. If it differs then it throws an error there rather than allowing it to propagate down-tree.

So what we'd want to do actually is write a custom error extension to print out the exact path etc, right now since this just throws a normal runtime error you'd get say this (in this case I took the response and omitted `name` which is considered non-nullable currently):

![](http://static.damonzucconi.com/_capture/t4VxkFk6Ew59.png)

Reply:
Or let's say Gravity was feeling creative and decided to put a number there:

![](http://static.damonzucconi.com/_capture/vhatg4ostA1P.png)

Ultimately in practice though what we'd see is errors in Sentry that are much simpler to debug rather than filtering out to different resolvers. We'd simply update the typing to be accurate and fix the bug with this newfound knowledge 🧠

Reply:
Makes sense -- ty 👍